### ISSUE TITLE
Nested slot become plain element

### ISSUE THREAD

**Issue by matthewma7** [vuejs/vue#8328]
### Version
2.5.16

### Reproduction link
[https://jsfiddle.net/jfmorio/qp6ra20u/](https://jsfiddle.net/jfmorio/qp6ra20u/)

### Steps to reproduce
execute the JSFiddle

### What is expected?
the nested named slot will be recognized and rendered inside the &#x3C;div class=&#x22;named-slot-container&#x22;&#x3E; in the end

### What is actually happening?
the nested named slot rendered as a plain element as default slot

---
Basically, slot needs to be defined at the root level. Maybe this is not possible with the current design.

A workaround is to define slot ONLY at the root level. https://jsfiddle.net/jfmorio/4pbno6je/

**Comment by Justineo** [comment]
This is the expected behavior. Slots inside a component are exposed to only its parent. If you want to specify slot content outside the parent, you should define a named slot in the parent that rendered as the corresponding child slot.
